async_firebase/errors.py

```diff
@@ -39,7 +39,7 @@ def __init__(
         self,
         code: str,
         message: str,
-        cause: t.Optional[Exception] = None,
+        cause: t.Union[httpx.HTTPStatusError, httpx.RequestError, None] = None,
```

Comment:
Just a nitpick, this is the same as `t.Optional[httpx.HTTPStatusError, httpx.RequestError]`, and does `Optional` make the use of the `cause` attribute more convenient? Maybe it could be changed to ` t.Union[httpx.HTTPStatusError, httpx.RequestError]` if mypy doesn't complain of course.

Reply:
`t.Optional` may not have multiple arguments, so only `t.Union` works in this case.